File: app/src/main/java/com/stepyen/xlearn/fragment/utils/launch/AActivity.java
package com.stepyen.xlearn.fragment.utils.launch;

import android.content.Intent;
import android.util.Log;
import android.view.View;
import android.widget.TextView;
import com.stepyen.xlearn.base.BaseTestActivity;

/**
 * date：2019/4/4
 * author：stepyen
 * description：
 */
public class AActivity extends BaseTestActivity {
    private static final String TAG = "AActivity";
    @Override
    public void initView() {
        Log.i(TAG, "onCreate: ");

        TextView tv = new TextView(this);
        tv.setText("AActivity");
        addView(tv);

        addView("BActivity", new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(AActivity.this, BActivity.class));
            }
        });
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.i(TAG, "onStart: ");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.i(TAG, "onResume: ");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.i(TAG, "onPause: ");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.i(TAG, "onStop: ");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.i(TAG, "onDestroy: ");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.i(TAG, "onRestart: ");
    }
}

File: app/src/main/java/com/stepyen/xlearn/activity/module/task/AActivity.kt
package com.stepyen.xlearn.activity.module.task

import android.content.Intent
import android.view.View
import com.stepyen.common.BaseLifePageActivity
import com.stepyen.xlearn.base.BasePageActivity

/**
 * date：2020-03-13
 * author：stepyen
 * description：
 *
 */
class AActivity : BaseLifePageActivity() {
    override var TAG =  "A_TAG"

    override fun initView() {
        addTextView("当前是 A")
        addButton("B",View.OnClickListener {
            startActivity(Intent(this@AActivity,BActivity::class.java).apply {

            })
        })
        addButton("BSingleTop",View.OnClickListener {
            startActivity(Intent(this@AActivity,BSingleTopActivity::class.java).apply {

            })
        })
        addButton("BSingleTask",View.OnClickListener {
            startActivity(Intent(this@AActivity,BSingleTaskActivity::class.java).apply {

            })
        })
        addButton("BSingleInstance",View.OnClickListener {
            startActivity(Intent(this@AActivity,BSingleInstanceActivity::class.java).apply {

            })
        })

        addButton("BBTask",View.OnClickListener {
            startActivity(Intent(this@AActivity,BBTaskActivity::class.java).apply {
                flags = Intent.FLAG_ACTIVITY_NEW_TASK
            })
        })
    }

}